### DISCUSSION TITLE
Websocket API example for C or C++

### DISCUSSION THREAD

**Discussion by peteallard** [yamcs/yamcs#756] [Q&A] [answered]
HI, I am trying to write a c++ client that will use the websocket API to get connected to a YAMCS server. 

I have searched the Web without success. Is there a C or C++ YAMCS client out there that I could start from ?

Thanks !

Pete

**Comment by fqqb** [comment, marked as answer]
Not for those languages (only Java, Python and JavaScript).
But since it's just JSON over WebSocket, you could use a general WebSocket client like the one from Boost [1], and then see [2] as an example JSON communication flow. For other subscriptions, there's documentation on the JSON structure on each separate page.

Note that not all methods are available over WebSocket -- only subscriptions. So depending on what you need, you may need to combine with an HTTP client.

[1] https://www.boost.org/doc/libs/release/libs/beast/doc/html/beast/quick_start/websocket_client.html
[2] https://docs.yamcs.org/yamcs-http-api/websocket/#example

**Reply by peteallard** [reply]
Thanks for your answer, it will definitely help !
